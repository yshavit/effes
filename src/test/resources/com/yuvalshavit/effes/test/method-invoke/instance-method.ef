type Dog:
  isCute -> True: return True

isDogCute (d: Dog) -> True | False:
  return d.isCute

main:
  debugPrint (isDogCute Dog)
