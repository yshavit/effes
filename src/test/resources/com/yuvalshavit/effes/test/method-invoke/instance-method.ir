def isDogCute (d: Dog) -> False | True:
  ReturnStatement
    MethodInvoke
      method: Dog::isCute
def Dog::isCute () -> True:
  ReturnStatement
    CtorInvoke
      type: True
