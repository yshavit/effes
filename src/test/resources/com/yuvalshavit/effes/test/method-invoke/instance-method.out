Your argument was: True
>> null
